di-ipv-ticf-stub/lambdas/src/services/ticfService.ts

```diff
@@ -1,98 +1,53 @@
+import TicfVc from "../domain/ticfVc";
+import { signJwt } from "./signingService";
 
 export async function processGetVCRequest(
   ticfRequest: TicfRequest
 ): Promise<TicfResponse> {
-  // preparing response
-  let ticfSigningKey: string;
-  try {
-    ticfSigningKey = await getSsmParameter(
-      process.env.TICF_PARAM_BASE_PATH + "signingKey"
-    );
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    throw new Error(`Error while retrieving TicF CRI VC signing key. . Error message: ${error.message}`);
-  }
-  let ticfComponentId: string;
-  try {
-    ticfComponentId = await getSsmParameter(
-      process.env.TICF_PARAM_BASE_PATH + "componentId"
-    );
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    throw new Error(`Error while retrieving TicF CRI componentId for VC aud value. Error message: ${error.message}`);
-  }
-  let timeoutVC: string | null | undefined = await getSsmParameter(
-    process.env.TICF_PARAM_BASE_PATH + "timeoutVC"
-  );
-  timeoutVC ??= "false";
-  let includeCIToVC: string | null | undefined = await getSsmParameter(
-    process.env.TICF_PARAM_BASE_PATH + "includeCIToVC"
-  );
-  includeCIToVC ??= "false";
+  const ticfSigningKey = await getSsmParameter(process.env.TICF_PARAM_BASE_PATH + "signingKey");
+  const ticfComponentId = await getSsmParameter(process.env.TICF_PARAM_BASE_PATH + "componentId");
+  const timeoutVc = await getSsmParameter(process.env.TICF_PARAM_BASE_PATH + "timeoutVC") === 'true';
+  const includeCi = await getSsmParameter(process.env.TICF_PARAM_BASE_PATH + "includeCIToVC") === 'true';
 
-  const buildJwtParams: SignedJwtParams = {
-    issuer: process.env.ISSUER,
-    customClaims: getCustomClaims(
-      JSON.parse(timeoutVC.toLowerCase()),
-      JSON.parse(includeCIToVC.toLowerCase()),
-      ticfRequest.sub,
-      ticfComponentId
-    ),
-    privateSigningKey: ticfSigningKey!,
-  };
-
-  // preparing response
-  try {
-    const returnJwt = await buildSignedJwt(buildJwtParams);
-    return {
-      sub: ticfRequest.sub,
-      govuk_signin_journey_id: ticfRequest.govuk_signin_journey_id,
-      vtr: ticfRequest.vtr,
-      vot: ticfRequest.vot,
-      vtm: ticfRequest.vtm,
-      "https://vocab.account.gov.uk/v1/credentialJWT": [returnJwt],
-    };
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    throw new Error(`Error while building signing JWT. Error message: ${error.message}`);
-  }
-}
+  const timestamp = Math.floor(new Date().getTime() / 1000);
 
-function getCustomClaims(
-  timeOutVC: boolean,
-  includeCIToVC: boolean,
-  userId: string,
-  componentId: string
-): JWTPayload {
-  return {
-    sub: userId,
-    iss: process.env.ISSUER,
-    aud: componentId,
-    nbf: Date.now(),
+  const payload: TicfVc = {
+    iss: ticfComponentId,
+    sub: ticfRequest.sub,
+    aud: ticfComponentId,
```

Comment:
I'm not sure the audience should actually use the same value as the issuer, but I'm checking with David to see what the right thing to do here is.